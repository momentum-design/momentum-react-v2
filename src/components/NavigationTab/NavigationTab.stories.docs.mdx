The `<NavigationTab />` component. This component accepts multiple scopes of child-like props:

- `label` - The label associated with this component, like 'Messaging'.
- `icon` - The icon associated with this component, like 'Chat'.
- `size` - The size of this component. It can be 48 (compact version) or 200 (expanded version).
- `count` - The count included in the blue badge of this component. If 0, the badge is not shown.
  Otherwise, the count is shown inside the badge.
- `active` - True if the navigation tab is active.
