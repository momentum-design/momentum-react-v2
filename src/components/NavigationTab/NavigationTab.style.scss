.md-navigation-tab-wrapper {
  cursor: pointer;
  font-family: var(--md-globals-font-default);
  transition: background-color 0.2s, color 0.2s, border-color 0.2s;
  // ----- COLOR ----- //

  // Circle/Pill //
  border-radius: 6rem;
  outline: none !important;
  border-color: rgba(0, 0, 0, 0);
  background-color: var(--mds-color-theme-button-secondary-normal);

  > .md-navigation-tab-icon > * {
    color: var(--mds-color-theme-text-secondary-normal);
  }

  > .md-navigation-tab-label {
    color: var(--mds-color-theme-text-secondary-normal);
  }

  &:hover {
    background-color: var(--mds-color-theme-button-secondary-hover);

    > .md-navigation-tab-icon > * {
      color: var(--mds-color-theme-text-secondary-normal);
    }

    > .md-navigation-tab-label {
      color: var(--mds-color-theme-text-secondary-normal);
    }
  }

  &:active {
    background-color: var(--mds-color-theme-button-secondary-pressed);

    > .md-navigation-tab-icon > * {
      color: var(--mds-color-theme-text-primary-normal);
    }

    > .md-navigation-tab-label {
      color: var(--mds-color-theme-text-primary-normal);
    }
  }

  &[data-active='true'] {
    background-color: var(--mds-color-theme-button-secondary-pressed);

    @media (forced-colors: active) {
      background-color: SelectedItem;
      color: SelectedItemText;
    }

    > .md-navigation-tab-icon > * {
      color: var(--mds-color-theme-text-primary-normal);
    }

    > .md-navigation-tab-label {
      color: var(--mds-color-theme-text-primary-normal);
    }

    &:hover {
      background-color: var(--mds-color-theme-button-secondary-pressed);

      > .md-navigation-tab-icon > * {
        color: var(--mds-color-theme-text-primary-normal);
      }

      > .md-navigation-tab-label {
        color: var(--mds-color-theme-text-primary-normal);
      }
    }

    &:active {
      background-color: var(--mds-color-theme-button-secondary-pressed);

      > .md-navigation-tab-icon > * {
        color: var(--mds-color-theme-text-primary-normal);
      }

      > .md-navigation-tab-label {
        color: var(--mds-color-theme-text-primary-normal);
      }
    }
  }

  // ----- SIZING & ALIGNMENT ----- //

  > .md-navigation-tab-label {
    padding-left: 1rem;
    padding-right: 2rem;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  display: flex;
  position: relative;
  align-items: center;

  &[data-size='40'] {
    height: 2.5rem;
    width: 2.5rem;
    justify-content: center;

    > .md-navigation-tab-count {
      position: absolute;
      right: -0.375rem;
      top: 0;
    }
  }

  &[data-size='48'] {
    height: 3rem;
    width: 3rem;
    justify-content: center;

    > .md-navigation-tab-count {
      position: absolute;
      left: 1.4rem;
      top: 0.25rem;
    }
  }

  &[data-size='200'] {
    height: 2.5rem;
    width: 100%;

    > .md-navigation-tab-icon {
      padding-left: 0.2rem;
    }

    > .md-navigation-tab-count {
      position: absolute;
      right: 0.75rem;
    }
  }
}
