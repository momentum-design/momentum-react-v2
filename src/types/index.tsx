import type { Button } from '@momentum-design/components/dist/react';

export type MdcButtonProps = React.ComponentProps<typeof Button>;
