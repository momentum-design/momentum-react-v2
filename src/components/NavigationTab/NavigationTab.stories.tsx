import { MultiTemplate, Template } from '../../storybook/helper.stories.templates';
import { DocumentationPage } from '../../storybook/helper.stories.docs';
import StyleDocs from '../../storybook/docs.stories.style.mdx';

import NavigationTab, { NavigationTabProps } from './';
import argTypes from './NavigationTab.stories.args';
import Documentation from './NavigationTab.stories.docs.mdx';

export default {
  title: 'Momentum UI/NavigationTab',
  component: NavigationTab,
  parameters: {
    expanded: true,
    docs: {
      page: DocumentationPage(Documentation, StyleDocs),
    },
  },
  args: {
    icon: 'chat',
  },
};

const Example = Template<NavigationTabProps>(NavigationTab).bind({});

Example.argTypes = { ...argTypes };

const Sizes = MultiTemplate<NavigationTabProps>(NavigationTab).bind({});

Sizes.argTypes = { ...argTypes };
delete Sizes.argTypes.size;

Sizes.parameters = {
  variants: [{ size: undefined }, { size: 40 }, { size: 48 }, { size: 200 }],
};

const Active = MultiTemplate<NavigationTabProps>(NavigationTab).bind({});

Active.argTypes = { ...argTypes };
delete Active.argTypes.active;

Active.parameters = {
  variants: [{ active: undefined }, { active: true }, { active: false }],
};

export { Example, Sizes, Active };
