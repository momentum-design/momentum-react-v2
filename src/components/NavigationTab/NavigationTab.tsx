import React, { FC } from 'react';
import classnames from 'classnames';
import FocusRing from '../FocusRing';

import Icon from '../Icon';
import Badge from '../Badge';
import Text from '../Text';
import ButtonSimple from '../ButtonSimple';

import { DEFAULTS, STYLE } from './NavigationTab.constants';
import { Props } from './NavigationTab.types';
import './NavigationTab.style.scss';

const NavigationTab: FC<Props> = (props: Props) => {
  const { icon, label, count = 0, className, id, size, style, active, ...otherProps } = props;

  const iconComponent = icon ? (
    <Icon className={STYLE.icon} name={icon} scale={24} weight={'filled'} strokeColor={'none'} />
  ) : null;

  const labelComponent =
    size == 200 && label ? (
      <Text className={STYLE.label} type="subheader-secondary" tagName="h3">
        {label}
      </Text>
    ) : null;

  //If the count is 100 or more, the count displayed should be :).
  const countToDisplay = (n: number) => {
    return n < 100 ? n.toString() : '+99';
  };

  const badgeComponent =
    count > 0 ? (
      <Badge className={STYLE.count} size={18}>
        {countToDisplay(count) || DEFAULTS.COUNT.toString()}
      </Badge>
    ) : null;

  return (
    <FocusRing>
      <ButtonSimple
        className={classnames(STYLE.wrapper, className)}
        data-active={active || DEFAULTS.ACTIVE}
        data-size={size || DEFAULTS.SIZE}
        id={id}
        style={style}
        {...otherProps}
      >
        {iconComponent}
        {labelComponent}
        {badgeComponent}
      </ButtonSimple>
    </FocusRing>
  );
};

export default NavigationTab;
