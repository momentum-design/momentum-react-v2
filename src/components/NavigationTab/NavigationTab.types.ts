import { CSSProperties } from 'react';
import { AriaButtonProps } from '@react-types/button';
import { InferredIconName } from '../Icon/Icon.types';

export interface Props extends AriaButtonProps {
  /**
   * Custom class for overriding this component's CSS.
   */
  className?: string;

  /**
   * Custom id for overriding this component's CSS.
   */
  id?: string;

  /**
   * Custom style for overriding this component's CSS.
   */
  style?: CSSProperties;

  /**
   * Size index of this NavTab.
   */
  size?: number;

  /**
   * Size index of this NavTab.
   */
  label?: string;

  /**
   * Size index of this NavTab.
   */
  icon?: InferredIconName;

  /**
   * The amount inside the badge of components of this NavTab. If 0, then it is not shown.
   */
  count?: number;

  /**
   * True if the tab is active.
   */
  active?: boolean;
}

export type NavTabSize = 40 | 48 | 200;
