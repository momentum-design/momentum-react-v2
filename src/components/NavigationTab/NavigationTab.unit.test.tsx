import React from 'react';
import { mount } from 'enzyme';

import NavigationTab, { NAVIGATION_TAB_CONSTANTS as CONSTANTS } from './';
import { STYLE, SIZES } from './NavigationTab.constants';
import { NavTabSize } from './NavigationTab.types';
import { mountAndWait } from '../../../test/utils';

describe('<NavigationTab />', () => {
  describe('snapshot', () => {
    it('should match snapshot', async () => {
      expect.assertions(1);

      const container = await mountAndWait(<NavigationTab aria-label="chat" />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', async () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = await mountAndWait(
        <NavigationTab aria-label="chat" className={className} />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', async () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = await mountAndWait(<NavigationTab aria-label="chat" id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', async () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const container = await mountAndWait(<NavigationTab aria-label="chat" style={style} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with size', async () => {
      expect.assertions(1);

      const sizes = Object.values(SIZES).map((size, index) => {
        return <NavigationTab aria-label="chat" key={index} size={size as NavTabSize} />;
      });
      const container = await mountAndWait(<div>{sizes}</div>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with label', async () => {
      expect.assertions(1);

      const label = 'Example label';

      const container = await mountAndWait(<NavigationTab label={label} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with icon', async () => {
      expect.assertions(1);

      const icon = 'contacts';

      const container = await mountAndWait(<NavigationTab aria-label="chat" icon={icon} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with count', async () => {
      expect.assertions(1);

      const count = 100;

      const container = await mountAndWait(<NavigationTab aria-label="chat" count={count} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with active', async () => {
      expect.assertions(1);

      const active = true;

      const container = await mountAndWait(<NavigationTab aria-label="chat" active={active} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with size, label, icon', async () => {
      expect.assertions(1);

      const size = 48;
      const label = 'Contacts';
      const icon = 'contacts';

      const container = await mountAndWait(<NavigationTab size={size} label={label} icon={icon} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with size, label, icon, count, active', async () => {
      expect.assertions(1);

      const size = 48;
      const label = 'Contacts';
      const icon = 'contacts';
      const count = 0;
      const active = false;

      const container = await mountAndWait(
        <NavigationTab size={size} label={label} icon={icon} count={count} active={active} />
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its wrapper class', async () => {
      expect.assertions(1);

      const wrapper = await mountAndWait(<NavigationTab aria-label="chat" />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      expect(element.classList.contains(STYLE.wrapper)).toBe(true);
    });

    it('should have provided class when className is provided', async () => {
      expect.assertions(1);

      const className = 'example-class';

      const wrapper = await mountAndWait(<NavigationTab aria-label="chat" className={className} />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', async () => {
      expect.assertions(1);

      const id = 'example-id';

      const wrapper = mount(<NavigationTab aria-label="chat" id={id} />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', async () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      const wrapper = await mountAndWait(<NavigationTab aria-label="chat" style={style} />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });

    it('should have provided size when size is provided', async () => {
      expect.assertions(1);

      const size = CONSTANTS.DEFAULTS.SIZE;

      const wrapper = await mountAndWait(<NavigationTab aria-label="chat" size={size} />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      expect(element.getAttribute('data-size')).toBe(`${size}`);
    });

    it('should have provided label when label is provided and size is 200 (size 48 has no label)', async () => {
      expect.assertions(1);

      const label = 'Messaging';

      const wrapper = await mountAndWait(<NavigationTab size={200} label={label} />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      const target = element.getElementsByClassName(STYLE.label)[0];

      expect(target.innerHTML).toBe(label);
    });

    it('should have provided count when count is provided and is >0', async () => {
      expect.assertions(1);

      const count = 1;

      const wrapper = await mountAndWait(<NavigationTab aria-label="chat" count={count} />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      const target = element.getElementsByClassName(STYLE.count)[0];

      expect(parseInt(target.innerHTML)).toBe(count);
    });

    it('should have provided active when active is provided', async () => {
      expect.assertions(1);

      const active = CONSTANTS.DEFAULTS.ACTIVE;

      const wrapper = await mountAndWait(<NavigationTab aria-label="chat" active={active} />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      expect(element.getAttribute('data-active')).toBe(`${active}`);
    });
  });

  describe('actions', () => {
    it('should handle mouse press events', async () => {
      expect.assertions(1);

      const mockCallback = jest.fn();

      const wrapper = await mountAndWait(
        <NavigationTab aria-label="chat" onPress={mockCallback} />
      );
      const component = wrapper.find(NavigationTab);

      component.props().onPress({
        type: 'press',
        pointerType: 'mouse',
        shiftKey: false,
        ctrlKey: false,
        metaKey: false,
        target: component.getDOMNode(),
        altKey: false,
      });

      expect(mockCallback).toBeCalledTimes(1);
    });
  });
});
