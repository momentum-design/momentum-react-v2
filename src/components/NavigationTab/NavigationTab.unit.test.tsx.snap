// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationTab /> snapshot should match snapshot 1`] = `
<NavigationTab>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            />
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with active 1`] = `
<NavigationTab
  active={true}
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={true}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={true}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            />
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with className 1`] = `
<NavigationTab
  className="example-class"
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper example-class"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper example-class md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            />
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with count 1`] = `
<NavigationTab
  count={100}
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              <Badge
                className="md-navigation-tab-count"
                size={18}
              >
                <div
                  className="md-badge-wrapper md-navigation-tab-count"
                  data-size={18}
                >
                  +99
                </div>
              </Badge>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with icon 1`] = `
<NavigationTab
  icon="contacts"
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              <Icon
                className="md-navigation-tab-icon"
                name="contacts"
                scale={24}
                strokeColor="none"
                weight="filled"
              >
                <div
                  aria-hidden="true"
                  className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-navigation-tab-icon md-icon-no-shrink"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-autoscale={false}
                    data-scale={24}
                    data-test="contacts"
                    fill="currentColor"
                    height="100%"
                    style={
                      {
                        "stroke": "none",
                      }
                    }
                    viewBox="0, 0, 32, 32"
                    width="100%"
                  />
                </div>
              </Icon>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with id 1`] = `
<NavigationTab
  id="example-id"
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        id="example-id"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              id="example-id"
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            />
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with label 1`] = `
<NavigationTab
  label="Example label"
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            />
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with size 1`] = `
<div>
  <NavigationTab
    key="0"
    size={40}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <ButtonSimple
          className="md-navigation-tab-wrapper"
          data-active={false}
          data-size={40}
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <FocusRing>
            <FocusRing
              focusClass="md-focus-ring-wrapper children"
              focusRingClass="md-focus-ring-wrapper children"
            >
              <button
                className="md-navigation-tab-wrapper md-button-simple-wrapper"
                data-active={false}
                data-size={40}
                onBlur={[Function]}
                onClick={[Function]}
                onDragStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                type="button"
              />
            </FocusRing>
          </FocusRing>
        </ButtonSimple>
      </FocusRing>
    </FocusRing>
  </NavigationTab>
  <NavigationTab
    key="1"
    size={48}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <ButtonSimple
          className="md-navigation-tab-wrapper"
          data-active={false}
          data-size={48}
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <FocusRing>
            <FocusRing
              focusClass="md-focus-ring-wrapper children"
              focusRingClass="md-focus-ring-wrapper children"
            >
              <button
                className="md-navigation-tab-wrapper md-button-simple-wrapper"
                data-active={false}
                data-size={48}
                onBlur={[Function]}
                onClick={[Function]}
                onDragStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                type="button"
              />
            </FocusRing>
          </FocusRing>
        </ButtonSimple>
      </FocusRing>
    </FocusRing>
  </NavigationTab>
  <NavigationTab
    key="2"
    size={200}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <ButtonSimple
          className="md-navigation-tab-wrapper"
          data-active={false}
          data-size={200}
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <FocusRing>
            <FocusRing
              focusClass="md-focus-ring-wrapper children"
              focusRingClass="md-focus-ring-wrapper children"
            >
              <button
                className="md-navigation-tab-wrapper md-button-simple-wrapper"
                data-active={false}
                data-size={200}
                onBlur={[Function]}
                onClick={[Function]}
                onDragStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                type="button"
              />
            </FocusRing>
          </FocusRing>
        </ButtonSimple>
      </FocusRing>
    </FocusRing>
  </NavigationTab>
</div>
`;

exports[`<NavigationTab /> snapshot should match snapshot with size, label, icon 1`] = `
<NavigationTab
  icon="contacts"
  label="Contacts"
  size={48}
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              <Icon
                className="md-navigation-tab-icon"
                name="contacts"
                scale={24}
                strokeColor="none"
                weight="filled"
              >
                <div
                  aria-hidden="true"
                  className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-navigation-tab-icon md-icon-no-shrink"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-autoscale={false}
                    data-scale={24}
                    data-test="contacts"
                    fill="currentColor"
                    height="100%"
                    style={
                      {
                        "stroke": "none",
                      }
                    }
                    viewBox="0, 0, 32, 32"
                    width="100%"
                  />
                </div>
              </Icon>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with size, label, icon, count, active 1`] = `
<NavigationTab
  active={false}
  count={0}
  icon="contacts"
  label="Contacts"
  size={48}
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              <Icon
                className="md-navigation-tab-icon"
                name="contacts"
                scale={24}
                strokeColor="none"
                weight="filled"
              >
                <div
                  aria-hidden="true"
                  className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-navigation-tab-icon md-icon-no-shrink"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-autoscale={false}
                    data-scale={24}
                    data-test="contacts"
                    fill="currentColor"
                    height="100%"
                    style={
                      {
                        "stroke": "none",
                      }
                    }
                    viewBox="0, 0, 32, 32"
                    width="100%"
                  />
                </div>
              </Icon>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with style 1`] = `
<NavigationTab
  style={
    {
      "color": "pink",
    }
  }
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
        style={
          {
            "color": "pink",
          }
        }
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                {
                  "color": "pink",
                }
              }
              type="button"
            />
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;
